simulation_data.R

```diff
@@ -2,161 +2,109 @@
 library(R.utils)
-sourceDirectory('R/')
+library(raster)
+library(sp)
 
+R.utils::sourceDirectory('R/')
 
-#### Setup
-dst <- ""  # directory to store output datasets
+set.seed(12)
 
+# Prism principal components ----------------------------------------------
 
-#### Simulation parameters
-n_sims <- 25
 agg_factor <- 10
-
-
-#### Prism principal components
 caPr <- load_prism_pcs()
-caPr.disc <- aggregate(caPr, fact=agg_factor)
+caPr.disc <- aggregate(caPr, fact = agg_factor)
 print(n_values(caPr.disc[[1]]))
 plot(caPr.disc)
 
+# Simulate data function --------------------------------------------------
 
-#### Level: Low preferential sampling
-# Median preferential sampling contribution: 19.87%
-prevalences <- c()
-ps_contribs <- c()
-obs_cells <- c()
-n_specimen <- c()
-
-Alpha.case <- 0.5
-Alpha.ctrl <- 0.3
-beta.case <- c(1, 0.75, 0.25)
-beta.ctrl <- c(3, 1, 0.5)
-beta.loc <- c(-1.5, 1, -0.25)
-for (i in 1:n_sims){
-  
-  #### Simulate gaussian process
-  Theta <- 6
-  Phi <- 12
-  cells.all <- c(1:ncell(caPr.disc))[!is.na(values(caPr.disc[[1]]))]
-  coords <- xyFromCell(caPr.disc, cell=cells.all)
-  d <- as.matrix(dist(coords, diag=TRUE, upper=TRUE))
-  Sigma <- Exponential(d, range=Theta, phi=Phi)
-  W <- mvrnorm(n=1, mu=rep(0, length(cells.all)), Sigma)
-  N <- length(W)
-  
-  #### Simulate locations
-  locs <- simLocations(caPr.disc, beta.loc, w=W)
-  obs_cells <- c(obs_cells, sum(locs$status))
-  
-  ps_contribs <- c(ps_contribs, 
-                   calc_ps_contribution(caPr.disc, locs, beta.case, Alpha.case, beta.ctrl, Alpha.ctrl, W))
+simulate_data <- function(label, disc, n_sims=25, save_dir=NULL, 
+                          Alpha.case=NULL, Alpha.ctrl=NULL, 
+                          beta.case=NULL, beta.ctrl=NULL, beta.loc=NULL,
+                          Theta=6, Phi=12){
   
-  #### Simulate counts given locations
-  case.data <- simCounts(caPr.disc, locs, beta.case, Alpha.case, W)
-  ctrl.data <- simCounts(caPr.disc, locs, beta.ctrl, Alpha.ctrl, W)
-  prevalences <- c(prevalences, sum(case.data$y)/sum(case.data$y + ctrl.data$y))
-  n_specimen <- c(n_specimen, sum(case.data$y + ctrl.data$y))
-  
-  params <- list(
-    sampling="low",
-    beta.case=beta.case,
-    beta.ctrl=beta.ctrl,
-    alpha.case=Alpha.case,
-    alpha.ctrl=Alpha.ctrl,
-    beta.loc=beta.loc,
-    Theta=Theta,
-    Phi=Phi,
-    W=W
-  )
-  
-  if (dst != ""){
-    save_output(params, paste("params_low_", i, ".json", sep=""), dst=dst)
-    data <- list(
-      case.data=case.data,
-      ctrl.data=ctrl.data,
-      locs=locs
-    )
-    save_output(data, paste("data_low_", i, ".json", sep=""), dst=dst)
+  if (label == "low") {
+    Alpha.case <- 0.5
+    Alpha.ctrl <- 0.3
+    beta.case <- c(1, 0.75, 0.25)
+    beta.ctrl <- c(3, 1, 0.5)
+    beta.loc <- c(-1.5, 1, -0.25)
+  } else if (label == "high") {
+    Alpha.case <- 1
+    Alpha.ctrl <- -0.25
+    beta.case <- c(-1.5, 0.25, 0.25)
+    beta.ctrl <- c(3, 1, 0.5)
+    beta.loc <- c(-1.5, 1, -0.25)
```

Comment:
I just copied over the settings you had for the low and high scenarios. I think it'd be good to add some documentation here about what each parameter represents and which direction leads to more or less preferential sampling.

Reply:
added some comments on this.